modules/openxr/scene/openxr_composition_layer_cylinder.cpp

```diff
@@ -131,14 +133,19 @@ Ref<Mesh> OpenXRCompositionLayerCylinder::_create_fallback_mesh() {
 void OpenXRCompositionLayerCylinder::_notification(int p_what) {
 	switch (p_what) {
 		case NOTIFICATION_LOCAL_TRANSFORM_CHANGED: {
-			Transform3D transform = get_transform();
-			Quaternion quat(transform.basis.orthonormalized());
-			composition_layer.pose.orientation = { (float)quat.x, (float)quat.y, (float)quat.z, (float)quat.w };
-			composition_layer.pose.position = { (float)transform.origin.x, (float)transform.origin.y, (float)transform.origin.z };
+			update_transform();
 		} break;
 	}
 }
 
+void OpenXRCompositionLayerCylinder::update_transform() {
+	Transform3D reference_frame = XRServer::get_singleton()->get_reference_frame();
+	Transform3D transform = reference_frame.inverse() * get_transform();
+	Quaternion quat(transform.basis.orthonormalized());
+	composition_layer.pose.orientation = { (float)quat.x, (float)quat.y, (float)quat.z, (float)quat.w };
+	composition_layer.pose.position = { (float)transform.origin.x, (float)transform.origin.y, (float)transform.origin.z };
+}
```

Comment:
The method implementations look identical; can this method be instead added to `openxr_composition_layer` so all these derived implementations can access it?

Reply:
Unfortunately, they are each using a different struct for `composition_layer`, so while the property has the same name, each struct has different offsets.

I suppose we could do something like:

```
composition_layer.pose = get_openxr_pose();
```

... where `get_openxr_pose()` is a protected method on `OpenXRCompositionLayer` (the base class)

Reply:
I've added a `get_openxr_pose()` method in my latest push, per David's idea.